lib/metadata/bucketfile/backend.js

```diff
@@ -29,152 +21,41 @@ class BucketFileInterface {
     constructor() {
         this.logger = logger;
         if (cluster.isMaster) {
-            this.startServer();
-        } else {
-            this.refcnt = 0;
-            this.waitreco = 0;
-            this.realReConnect();
-            this.notifier = new events.EventEmitter();
+            this.mdServer = new MetadataServer(
+            { metadataPath: METADATA_PATH,
+              metadataPort: METADATA_PORT,
+              log: config.log });
+            this.mdServer.startServer();
+        }
+        this.mdClient = new MetadataClient(
+            { metadataHost: 'localhost',
+              metadataPort: METADATA_PORT,
+              log: config.log });
+        this.mdDB = this.mdClient.openDB();
+        this.metastore = this.mdDB.openSub(METASTORE);
+        if (cluster.isMaster) {
```

Comment:
as discussed, i was just commenting on the fact that the same condition is on line 23